src/spdx/parser/rdf/snippet_parser.py

```diff
@@ -0,0 +1,86 @@
+                                                                                         POINTER_NAMESPACE.offset))
+    line_range = apply_parsing_method_or_log_error(logger, POINTER_NAMESPACE.LineCharPointer,
+                                                   parsing_method=lambda x: parse_ranges(snippet_node, graph, x,
+                                                                                         POINTER_NAMESPACE.lineNumber))
+    license_concluded = parse_literal_or_no_assertion_or_none(
+        logger, graph, snippet_node, SPDX_NAMESPACE.licenseConcluded,
+        parsing_method=lambda x: parse_license_expression(x, graph, doc_namespace))
+    license_info_in_snippet = []
+    for (_, _, license_info_in_snippet_node) in graph.triples(
+        (snippet_node, SPDX_NAMESPACE.licenseInfoInSnippet, None)):
+        license_info_in_snippet.append(
+            get_correctly_typed_value(logger, license_info_in_snippet_node,
+                                      lambda x: parse_license_expression(x, graph, doc_namespace)))
+    license_comment = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.licenseComments)
+    copyright_text = parse_literal_or_no_assertion_or_none(logger, graph, snippet_node, SPDX_NAMESPACE.copyrightText)
+    comment = parse_literal(logger, graph, snippet_node, RDFS.comment)
+    name = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.name)
+    attribution_texts = []
+    for (_, _, attribution_text_literal) in graph.triples((snippet_node, SPDX_NAMESPACE.attributionText, None)):
+        attribution_texts.append(attribution_text_literal.toPython())
+
+    raise_parsing_error_if_logger_has_messages(logger, "Snippet")
+    snippet = construct_or_raise_parsing_error(Snippet,
+                                               dict(spdx_id=spdx_id, file_spdx_id=file_spdx_id, byte_range=byte_range,
+                                                    line_range=line_range, license_concluded=license_concluded,
+                                                    license_info_in_snippet=license_info_in_snippet,
+                                                    license_comment=license_comment,
+                                                    copyright_text=copyright_text, comment=comment, name=name,
+                                                    attribution_texts=attribution_texts))
+    return snippet
+
+
+def parse_ranges(snippet_node: URIRef, graph: Graph, pointer_class: URIRef, member: Node) -> Tuple[int, int]:
+    range_values = dict()
+    for (_, _, start_end_pointer) in graph.triples((snippet_node, SPDX_NAMESPACE.range, None)):
+        for pointer_property in [POINTER_NAMESPACE.startPointer, POINTER_NAMESPACE.endPointer]:
+            try:
+                pointer_node = graph.value(start_end_pointer, pointer_property, any=False)
+            except UniquenessError:
+                raise SPDXParsingError(["Multiple values for startPointer"])
```

Comment:
This loop also covers the endPointer, so the error message has to be more general.

Reply:
In fact this loop fails for both ranges if one of them has multiple start- or endPointer. I think with this I'm in favor of returning to the first implementation and iterating through the values.

Reply:
But then we disregard possible duplications, don't we?

Reply:
I'm still against the idea of using a for-loop as an if-condition, but if there is no other way...